mycroft/skills/skill_manager.py

```diff
@@ -342,20 +383,38 @@ def _remove_git_locks(self):
                 LOG.warning('Found and removed git lock file: ' + i)
                 os.remove(i)
 
+    def _load_on_network(self):
+        LOG.info('Loading network skills...')
+        self._load_new_skills(network=True, internet=False)
+
+    def _load_on_internet(self):
+        LOG.info('Loading internet skills...')
+        self._load_new_skills(network=True, internet=True)
+
     def _load_on_startup(self):
         """Handle initial skill load."""
-        self.load_plugin_skills()
-        LOG.info('Loading installed skills...')
-        self._load_new_skills()
-        LOG.info("Skills all loaded!")
-        self.bus.emit(Message('mycroft.skills.initialized'))
+        LOG.info('Loading offline skills...')
+        self._load_new_skills(network=False, internet=False)
```

Comment:
Should check network/internet connection here and set events in case the event came before the listener was registered